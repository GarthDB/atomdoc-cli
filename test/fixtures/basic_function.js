/**
 * Private: checks if a node is a decendant of an [AtRule](http://api.postcss.org/AtRule.html).
 *
 * * `node` {Object} PostCSS Node to check.
 *
 * ## Examples
 *
 * ```js
 * const atRule = postcss.parse('@media (min-width: 480px) {a{}}').first;
 * const rule = atRule.first;
 * _isAtruleDescendant(rule); // returns '(min-width: 480px)'
 * ```
 *
 * Returns {Boolean} of false, or {String} of AtRule params if true.
 */
function _isAtruleDescendant(node) {
  let { parent } = node;
  let descended = false;

  while (parent && parent.type !== 'root') {
    if (parent.type === 'atrule') {
      descended = parent.params;
    }
    parent = parent.parent;
  }
  return descended;
}
