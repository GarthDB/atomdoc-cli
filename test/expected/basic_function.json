[
  {
    "originalText": "Private: checks if a node is a decendant of an [AtRule](http://api.postcss.org/AtRule.html).\n\n* `node` {Object} PostCSS Node to check.\n\n## Examples\n\n```js\nconst atRule = postcss.parse('@media (min-width: 480px) {a{}}').first;\nconst rule = atRule.first;\n_isAtruleDescendant(rule); // returns '(min-width: 480px)'\n```\n\nReturns {Boolean} of false, or {String} of AtRule params if true.",
    "visibility": "Private",
    "description": "checks if a node is a decendant of an [AtRule](http://api.postcss.org/AtRule.html).",
    "summary": "checks if a node is a decendant of an [AtRule](http://api.postcss.org/AtRule.html).",
    "arguments": [
      {
        "name": "node",
        "description": "{Object} PostCSS Node to check.",
        "type": "Object",
        "isOptional": false
      }
    ],
    "examples": [
      {
        "description": "",
        "lang": "js",
        "code": "const atRule = postcss.parse('@media (min-width: 480px) {a{}}').first;\nconst rule = atRule.first;\n_isAtruleDescendant(rule); // returns '(min-width: 480px)'",
        "raw": "```js\nconst atRule = postcss.parse('@media (min-width: 480px) {a{}}').first;\nconst rule = atRule.first;\n_isAtruleDescendant(rule); // returns '(min-width: 480px)'\n```"
      }
    ],
    "returnValues": [
      {
        "type": "Boolean",
        "description": "Returns {Boolean} of false, or {String} of AtRule params if true."
      }
    ]
  }
]
